pkg/api/v1/types.go

```diff
@@ -1780,7 +1780,7 @@ type LoadBalancerIngress struct {
 type ServiceSpec struct {
 	// The list of ports that are exposed by this service.
 	// More info: http://releases.k8s.io/HEAD/docs/user-guide/services.md#virtual-ips-and-service-proxies
-	Ports []ServicePort `json:"ports"`
+	Ports []ServicePort `json:"ports" patchStrategy:"merge" patchMergeKey:"port"`
```

Comment:
If someone did manage to workaround this problem and was using strategic merge patch, their patches should still work.